mirascope/core/mistral/tool.py

```diff
@@ -72,8 +72,7 @@ def from_tool_call(cls, tool_call: ToolCall) -> MistralTool:
         Args:
             tool_call: The Mistral tool call from which to construct this tool instance.
         """
-        model_json = tool_call.function.arguments
-        if isinstance(model_json, str):
-            model_json = jiter.from_json(model_json.encode())
-        model_json["tool_call"] = tool_call.model_dump()
+        model_json = {"tool_call": tool_call.model_dump()}
+        if args := tool_call.function.arguments and isinstance(args, str):
```

Comment:
you'll have to wrap this:

```python
if (args := tool_call.function.arguments) and isinstance(args, str):
```

Reply:
oh wait actually this is wrong, cause if `tool_call.function.arguments` is not a string I believe it could be a dictionary, in which case we'll need to handle it differently from string (like we did before by setting `model_json = tool_call.function.arguments` directly)

Reply:
@willbakst I'm still making friends with the walrus :)

Reply:
@willbakst updated it. lmk how that looks.